src/client/Microsoft.Identity.Client/ManagedIdentity/V2/Cache/Abstractions/IBindingCache.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2.Cache.Abstractions
+{
+    internal interface IBindingCache
+    {
+        void Cache(string identityKey, string tokenType,
+                    CertificateRequestResponse resp, string subject);
+
+        bool TryGet(string identityKey, string tokenType,
```

Comment:
why do you need a key and a token type? This object should only deal with keys. The caller can add the token type to the key. SRP